### ISSUE TITLE
Custom Checkbox Component used in analytics -> reports not accessible via keyboard, or with screen-reader

### ISSUE THREAD

**Issue by joelclimbsthings** [woocommerce/woocommerce-admin#6206]
**Describe the bug**
The checkboxes used to toggle the comparison periods in the analytics charts are not accessible. They're not semantically recognizable as checkboxes at all, so screen readers cannot tell if they're checked/unchecked. 

**To Reproduce**

- Enable a screen reader
- Use a keyboard to _tab_ to the pictured checkboxes within a report
- Use the `spacebar` to toggle the checkbox, and notice that the screen-reader cannot tell you whether it's checked or not.

**Screenshots**

![image](https://user-images.githubusercontent.com/444632/106065981-eb1d2b00-60b0-11eb-93dd-7a5ba2070a5c.png)

**Expected behavior**
The checkboxes should be structured in a way that allows the screen reader to recognize that it's a checkbox, and what it's current state is.
